OsmAnd/src/net/osmand/plus/CustomOsmandPlugin.java

```diff
@@ -398,33 +415,120 @@ public void loadResources() {
 		}
 	}
 
-	private Drawable getIconForFile(String path, Map<String, String> fileNames) {
-		for (Map.Entry<String, String> entry : fileNames.entrySet()) {
-			String value = entry.getValue();
-			if (value.startsWith("@")) {
-				value = value.substring(1);
+	public void updateSuggestedDownloads(List<SuggestedDownloadItem> items) {
+		suggestedDownloadItems = new ArrayList<>(items);
+	}
+
+	public void updateDownloadItems(List<WorldRegion> items) {
+		customRegions = new ArrayList<>(items);
+	}
+
+	private List<IndexItem> getMapsForType(LatLon latLon, DownloadActivityType type) {
+		try {
+			return DownloadResources.findIndexItemsAt(app, latLon, type);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		return Collections.emptyList();
+	}
+
+	private List<IndexItem> getMapsForType(List<String> names, DownloadActivityType type, int limit) {
+		return DownloadResources.findIndexItemsAt(app, names, type, false, limit);
+	}
+
+	public static String getLocalizedResFromMap(Context ctx, Map<String, String> localizedMap, String defVal) {
```

Comment:
getLocalizedResFromMap, jsonArrayToList, getLocalizedMapFromJson, writeStringListToJson, writeLocalizedMapToJson and similar should be moved to external class like JsonUtils